oauth-common/src/main/java/io/strimzi/kafka/oauth/jsonpath/Matcher.java

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright 2017-2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.jsonpath;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.util.Iterator;
+
+class Matcher {
+
+    private static final Logger log = LoggerFactory.getLogger(Matcher.class);
+
+    private final StatementNode parsed;
+
+    Matcher(StatementNode parsed) {
+        this.parsed = parsed;
+    }
+
+    /**
+     * Match the JSON object against the JsonPath filter query.
+     *
+     * @param json Jackson JsonObject to match
+     * @return true if the object matches the filter, false otherwise
+     */
+    public boolean match(JsonNode json) {
+        BooleanEvaluator eval = new BooleanEvaluator();
+        for (ExpressionNode expression : parsed.expressions) {
+
+            Logical logical = expression.getOp();
+            if (logical == Logical.AND && !eval.current) {
+                return false;
+            }
+            PredicateNode predicate = expression.getPredicate();
+            OperatorNode op = predicate.getOp();
+
+            try {
+                if (op == OperatorNode.EQ) {
+                    eval.update(logical, compareEquals(json, predicate));
+                } else if (op == OperatorNode.NEQ) {
+                    eval.update(logical, !compareEquals(json, predicate));
+                } else if (op == OperatorNode.GT) {
+                    eval.update(logical, compareGreaterThan(json, predicate));
+                } else if (op == OperatorNode.LTE) {
+                    eval.update(logical, !compareGreaterThan(json, predicate));
+                } else if (op == OperatorNode.LT) {
+                    eval.update(logical, compareLessThan(json, predicate));
+                } else if (op == OperatorNode.GTE) {
+                    eval.update(logical, !compareLessThan(json, predicate));
+                } else if (op == OperatorNode.IN) {
+                    eval.update(logical, containedIn(json, predicate));
+                } else if (op == OperatorNode.NIN) {
+                    eval.update(logical, !containedIn(json, predicate));
+                } else if (op == OperatorNode.MATCH_RE) {
+                    throw new RuntimeException("Not implemented");
+                } else if (op == OperatorNode.ANYOF) {
+                    throw new RuntimeException("Not implemented");
+                } else if (op == OperatorNode.NONEOF) {
+                    throw new RuntimeException("Not implemented");
```

Comment:
If these aren't supported they should be rejected by the parser.

Reply:
Still working on these ...